project_task_pull_request_state/models/project_task.py

```diff
@@ -0,0 +1,108 @@
+# Copyright Cetmix OU 2023
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0).
+from odoo import api, fields, models
+
+
+class ProjectTaskState(models.Model):
+    _inherit = "project.task"
+
+    pr_state = fields.Selection(
+        selection=lambda self: self._selection_pr_state(),
+        tracking=True,
+        copy=False,
+        string="PR State",
+        compute="_compute_pr_state",
+        readonly=False,
+    )
+
+    def _selection_pr_state(self):
+        """Function to select the state of the pull request"""
+        return [
+            ("open", "Open"),
+            ("draft", "Draft"),
+            ("merged", "Merged"),
+            ("closed", "Closed"),
+        ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_recs = super(ProjectTaskState, self).create(vals_list)
+
+        # Set pr state if URI is provided at creation
+        for new_rec, vals in zip(new_recs, vals_list):
```

Comment:
pr_state will be computed automatically upon creation. No need to call this function explicitly

Reply:
resolved